lib/web_ui/lib/src/engine/semantics/semantics.dart

```diff
@@ -451,51 +462,69 @@ abstract class PrimaryRoleManager {
 
   /// Adds generic focus management features, if applicable.
   void addFocusManagement() {
-    if (semanticsObject.isFocusable) {
-      addSecondaryRole(Focusable(semanticsObject));
-    }
+    addSecondaryRole(Role.focusable, (SemanticsObject semanticsObject) {
+      if (semanticsObject.isFocusable) {
+        return Focusable(semanticsObject);
+      }
+      return null;
+    });
```

Comment:
Looks like these factories are always the same and have no dependency on the `PrimaryRoleManager` instance. Can we put them in a global map:
```dart
final Map<Role, RoleManagerFactory> secondaryRoleFactories = <Role, RoleManagerFactory> {
  Role.focusable: (SemanticsObject obj) => obj.isFocusable ? Focusable(obj) : null,
  ...
};
```
or combine them in a switch expression somewhere (I prefer this to make sure we exhaust all cases for `Role`):
```dart
extension RoleFactoryExtension on Role {
  RoleManager? createManager() {
    return switch (this) {
      Role.focusable => this.isFocusable ? Focusable(this) : null,
      ...
    }
  }
}
```

And the primary role manager only maintains a `List<Role>` and uses it during `update()` to decide what secondary role managers to create.